kolibri/core/content/utils/channel_import.py

```diff
@@ -950,6 +949,23 @@ def import_channel_data(self):
                 time.time() - start
             )
         )
+        # Create ChannelMetadata object deleted during import
+        if import_ran:
+            ids = list(
+                ChannelMetadata.objects.filter(root__available=True)
+                .all()
+                .values_list("id", flat=True)
+            )
+            if len(ids) != len(id_order):
+                deleted_channel_id = [id for id in id_order if id not in ids][0]
+                deleted_channel = channel_order.filter(id=deleted_channel_id)[0]
+                ChannelMetadata.objects.update_or_create(**deleted_channel)
+                ChannelMetadata.objects.update(
+                    order=Case(
+                        *(When(id=uuid, then=i + 1) for i, uuid in enumerate(id_order))
+                    )
+                )
+
         return import_ran
 
     def run_and_annotate(self):
```

Comment:
I think the logic here could be more simply encoded here.

Before the line `import_ran = self.import_channel_data()`

You can do:

```
try:
    old_order = ChannelMetadata.objects.values("order").get(id=self.channel_id)["order"]
except ChannelMetadata.DoesNotExist:
    old_order = None
```

then if the import did run, you can do:

```
if old_order is not None:
    channel.order = old_order
```

This preserves the order across the operations and avoids having to load the orders for every channel into memory.

Reply:
You're absolutely correct! It was mentioned in the issue that it'd be best solved in the import_channel_data function so I ended up thinking exclusively within that scope. Thanks for this fix, I've updated it.